lib/app/features/wallet/views/pages/manage_nfts/providers/manage_nfts_provider.dart

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/features/wallet/model/network_type.dart';
+import 'package:ion/app/features/wallet/views/pages/manage_nfts/model/manage_nft_network_data.dart';
+import 'package:ion/app/features/wallet/views/pages/manage_nfts/providers/mock_data/manage_nfts_mock_data.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'manage_nfts_provider.g.dart';
+
+@Riverpod(keepAlive: true)
+class ManageNftNetworksNotifier extends _$ManageNftNetworksNotifier {
+  @override
+  AsyncValue<Set<ManageNftNetworkData>> build() {
+    return AsyncData<Set<ManageNftNetworkData>>(mockedManageNftsNetworkDataSet);
+  }
+
+  void selectNetwork({
+    required bool isSelected,
+    required NetworkType networkType,
+  }) {
+    final currentSet = state.value ?? {};
+
+    if (networkType == NetworkType.all &&
+        isSelected &&
+        currentSet.any((nftNetworkData) => !nftNetworkData.isSelected)) {
+      return;
+    }
+
+    final updatedNftsNetworksSet = networkType == NetworkType.all
+        ? currentSet.map((nftNetworkData) {
+            if (nftNetworkData.networkType == NetworkType.all) {
+              return nftNetworkData.copyWith(isSelected: !isSelected);
+            }
+            return nftNetworkData.copyWith(isSelected: false);
+          }).toSet()
+        : currentSet.map((nftNetworkData) {
+            if (nftNetworkData.networkType == NetworkType.all) {
+              return nftNetworkData.copyWith(isSelected: false);
+            } else if (nftNetworkData.networkType == networkType) {
+              return nftNetworkData.copyWith(isSelected: !isSelected);
+            }
+
+            return nftNetworkData;
+          }).toSet();
```

Comment:
any option to flatten this code, or extract into methods, so we don't have nested if statements? It decreased readability a lot

Reply:
Can leave TODO item, currently have more priority task with DB

Reply:
```suggestion
    final updatedNftsNetworksSet = networkType == NetworkType.all
        ? currentSet.map((nftNetworkData) {
            if (nftNetworkData.networkType == NetworkType.all) {
              return nftNetworkData.copyWith(isSelected: !isSelected);
            }
            return nftNetworkData.copyWith(isSelected: false);
          }).toSet()
        : currentSet.map((nftNetworkData) {
            if (nftNetworkData.networkType == NetworkType.all) {
              return nftNetworkData.copyWith(isSelected: false);
            } else if (nftNetworkData.networkType == networkType) {
              return nftNetworkData.copyWith(isSelected: !isSelected);
            }

            return nftNetworkData;
          }).toSet();
```